src/toolbox/ngm-geometry-drawer.ts

```diff
@@ -112,9 +91,24 @@ export class NgmAreaOfInterestDrawer extends LitElementI18n {
       this.increaseAreasCounter(conf.type);
       this.addAreaEntity(conf);
     });
-    QueryStore.objectInfo.subscribe(info => {
-      if (!info) {
+    DrawStore.draw.subscribe(draw => {
+      if (draw) {
+        this.draw = draw;
+        this.draw.addEventListener('statechanged', () => this.requestUpdate());
+        this.draw.addEventListener('drawend', (evt) => this.endDrawing_((<CustomEvent>evt).detail));
+        this.draw.addEventListener('drawerror', evt => {
```

Comment:
There are several places where you subscribe or add an event listener but I see no unlistening. Is it correct?

Reply:
we don't need to unlistening these events as we have only one example of geometry-drawer element and we should always listen for draw events